lib/rubocop/cop/lint/number_conversion.rb

```diff
@@ -76,13 +87,31 @@ def on_send(node)
           end
         end
 
-        private
+        def handle_to_symbol(node)
+          to_method_symbol(node) do |receiver, sym_node, to_method|
+            next if receiver.nil?
+
+            message = format(
+              MSG,
+              current: sym_node.source,
+              corrected_method: correct_sym_method(to_method)
+            )
+            add_offense(node, message: message) do |corrector|
+              corrector.replace(sym_node, correct_sym_method(to_method))
+            end
+          end
+        end
 
         def correct_method(node, receiver)
           format(CONVERSION_METHOD_CLASS_MAPPING[node.method_name],
                  number_object: receiver.source)
         end
 
+        def correct_sym_method(to_method)
+          body = format(CONVERSION_METHOD_CLASS_MAPPING[to_method], number_object: 'i')
+          "{ |i| #{body} }"
```

Comment:
I'd probably suggest here something like `&method(:Integer)` as it's a bit more compact and closer to their original code.

Reply:
For `Integer` we pass in base 10 as second argument. We cannot do that with `&method(:Integer)` right?

Reply:
Should we use explicit block for `Integer` and method-to-proc for `Float` and `Complex`?

Reply:
I guess it'd best to stay uniform with those. You can keep them as blocks.